src/reason_parser.mly

```diff
@@ -3196,6 +3196,13 @@ mark_position_pat
   | pattern_without_or AS as_loc(error)
     { expecting_pat (with_txt $3 "identifier") }
 
+  /* | Foo () => ... is sugar for | Foo (()) => ...*/
+  /* putting it here instead of in pattern_constructor_argument because that one might cause too many shift/reduce conflicts... */
```

Comment:
For the record: We *could* put it in pattern_constructor_argument, but it would require some refactoring to avoid the conflicts, and putting it here isn't so bad - since this `X()` being equivalent to `X(())` is a special cased sugar anyways.